state/factory/workingset.go

```diff
@@ -419,7 +425,7 @@ func (ws *workingSet) pickAndRunActions(
 				actionIterator.PopAccount()
 				continue
 			}
-			actionCtx, err := withActionCtx(ctxWithBlockContext, nextAction)
+			actionCtx, err := withActionCtx(ctxWithBlockContext, nextAction, txIndex, logIndex)
```

Comment:
Could the receipt be updated after line447 not inside runAction() with actionCtx?

Reply:
thx, it might be easier to update receipt/log together here, instead of change for every protocol
will check and udpate